dspace-api/src/main/java/org/dspace/app/suggestion/oaire/DateScorer.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.suggestion.oaire;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+import org.dspace.app.suggestion.SuggestionEvidence;
+import org.dspace.app.suggestion.SuggestionUtils;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.ItemService;
+import org.dspace.external.model.ExternalDataObject;
+import org.dspace.util.MultiFormatDateParser;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@see org.dspace.app.suggestion.oaire.EvidenceScorer} which evaluate ImportRecords
+ * based on the distance from a date extracted from the ResearcherProfile (birthday / graduation date)
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science dot it)
+ *
+ */
+public class DateScorer implements EvidenceScorer {
+
+    private String birthDateMetadata;
+
+    private String educationDateMetadata;
+
+    private String minDateMetadata;
+
+    private String maxDateMetadata;
+
+    private int birthDateDelta = 20;
+    private int birthDateRange = 50;
+
+    private int educationDateDelta = -3;
+    private int educationDateRange = 50;
```

Comment:
[see](https://github.com/DSpace/DSpace/pull/8280#discussion_r1466285610)

Reply:
Same here, we need to add notes into the code itself to describe the meaning behind "-3" and "50"?  Are these somehow matched against the year?  It should be described in the code so that other developers can understand their meaning in the future.

Reply:
javadoc has been added.